examples/columnar.rs

```diff
@@ -0,0 +1,453 @@
+        <CR as AsSpecificBytes>::Owned : Default + IsZero + for<'a> Semigroup<<CR::Borrowed<'a> as Index>::Ref>,
+        for<'a> CR::Borrowed<'a> : Len + Index,
+        CR: for<'a> Push<&'a <CR as AsSpecificBytes>::Owned>,
+    {
+        type TimeOwned = <CT as AsSpecificBytes>::Owned;
+        type DiffOwned = <CR as AsSpecificBytes>::Owned;
+
+        fn time_kept(item: &Self::Item<'_>, upper: &AntichainRef<Self::TimeOwned>, frontier: &mut Antichain<Self::TimeOwned>) -> bool {
+            let (_, time, _) = item;
+            if upper.less_equal(time) {
+                frontier.insert_with(time, |time| <CT as AsSpecificBytes>::into_owned(time));
+                true
+            } 
+            else { false }
+        }
+
+        fn push_and_add<'a>(&mut self, item1: Self::Item<'a>, item2: Self::Item<'a>, stash: &mut Self::DiffOwned) {
+            let (data, time, diff1) = item1;
+            let (___d, ___t, diff2) = item2;
+            <CR as AsSpecificBytes>::clone_onto(&diff1, stash);
+            stash.plus_equals(&diff2);
+            self.push_into((data, time, &*stash));
+        }
+
+        fn account(&self) -> (usize, usize, usize, usize) {
+            let (size, capacity, allocations) = (0, 0, 0);
+            (self.len(), size, capacity, allocations)
+        }
+    }
+
+    struct ColumnQueue<C> {
+        list: C,
+        head: usize,
+    }
+
+    use differential_dataflow::trace::implementations::merge_batcher::container::ContainerQueue;
+    impl<CD, CT, CR> ContainerQueue<Column<(CD, CT, CR)>> for ColumnQueue<Column<(CD, CT, CR)>> 
+    where 
+        CD: Default + Clone + Len + Index + Clear + AsBytes + for<'a> Push<<CD::Borrowed<'a> as Index>::Ref> + 'static,
+        for<'a> CD::Borrowed<'a> : Len + Index,
+        for<'a> <CD::Borrowed<'a> as Index>::Ref: Ord,
+        CT: Default + Clone + Len + Index + Clear + AsBytes + for<'a> Push<<CT::Borrowed<'a> as Index>::Ref> + 'static,
+        for<'a> CT::Borrowed<'a> : Len + Index,
+        for<'a> <CT::Borrowed<'a> as Index>::Ref: Ord,
+        CR: Default + Clone + Len + Index + Clear + AsBytes + for<'a> Push<<CR::Borrowed<'a> as Index>::Ref> + 'static,
+        for<'a> CR::Borrowed<'a> : Len + Index,
+    {
+        fn next_or_alloc(&mut self) -> Result<<Column<(CD, CT, CR)> as Container>::Item<'_>, Column<(CD, CT, CR)>> {
+            if self.is_empty() { Err(std::mem::take(&mut self.list)) }
+            else {
+                self.head += 1;
+                Ok(self.list.borrow().get(self.head - 1))
+            }
+        }
+        fn is_empty(&self) -> bool { self.head == self.list.len() }
+        fn cmp_heads(&self, other: &Self) -> std::cmp::Ordering {
+            let (data1, time1, _) = self.list.borrow().get(self.head);
+            let (data2, time2, _) = other.list.borrow().get(other.head);
+            (data1, time1).cmp(&(data2, time2))
+        }
+        fn from(list: Column<(CD, CT, CR)>) -> Self { Self { list, head: 0 } }
+    }
+}
```

Comment:
https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files :)

Reply:
What I need is a link to the doc that explains why I should care as much as the rest of the internet. :D